x/launch/types/message_request_add_vested_account.go

```diff
@@ -0,0 +1,75 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ sdk.Msg = &MsgRequestAddVestedAccount{}
+
+func NewMsgRequestAddVestedAccount(address, chainID string,
+	coins sdk.Coins, options *types.Any) *MsgRequestAddVestedAccount {
+	return &MsgRequestAddVestedAccount{
+		ChainID: chainID,
+		Address: address,
+		Coins:   coins,
+		Options: options,
+	}
+}
+
+func (msg *MsgRequestAddVestedAccount) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgRequestAddVestedAccount) Type() string {
+	return "RequestAddVestedAccount"
+}
+
+func (msg *MsgRequestAddVestedAccount) GetSigners() []sdk.AccAddress {
+	address, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{address}
+}
+
+func (msg *MsgRequestAddVestedAccount) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRequestAddVestedAccount) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Address)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid address (%s)", err)
+	}
+
+	_, _, err = ParseChainID(msg.ChainID)
+	if err != nil {
+		return sdkerrors.Wrap(ErrInvalidChainID, msg.ChainID)
+	}
+
+	if msg.Coins.Empty() {
```

Comment:
We can remove this check because we can actually have an empty starting balance: the account only has vesting coins.

The opposite is not true though because is there is no vesting coins then the user should use a regular genesis account

Reply:
great, thks!
should I remove this one too?

https://github.com/tendermint/spn/pull/222/files#diff-e3eb87deb7e77fead8c2e3e18dd83ce6edc6dc8d7071257c269dffc2a3d09b58R16-R18

Reply:
No we must have vesting coins otherwise there is no point using a vestedAccount instead of genesisAccount